### PR TITLE
runtimes/core+js: add details to errors

### PR DESCRIPTION
**Pull request by fredr**

### CODE REVIEW

File: runtimes/js/encore.dev/api/error.ts [hunk 1]
```diff
@@ -112,6 +114,8 @@ export class APIError extends Error {
   }
 }
 
+export type ErrDetails = object;
```
Comment (eandre) [inline runtimes/js/encore.dev/api/error.ts:117]: I think this should be `Record<string, any>`

File: runtimes/js/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+use crate::log::parse_js_stack;
+use encore_runtime_core::api;
+use napi::{Env, JsUnknown, Status};
+
+pub fn coerce_to_api_error(env: Env, val: napi::JsUnknown) -> Result<api::Error, api::Error> {
```
Comment (eandre) [inline runtimes/js/src/error.rs:5]: Should the return type just be `api::Error`? Is the Ok/Err distinction useful?

Comment (fredr) [reply runtimes/js/src/error.rs:5]: We can change it to that, its just nice to be able to know if the coerce succeeded or not imo, even though both success and fail is the same type.

As we use it today we dont care if it succeeded or not, so I dont have strong opinions about it

Comment (eandre) [reply runtimes/js/src/error.rs:5]: Fair, can keep it as-is

**Review by eandre — APPROVED** [review]

**Pull request merged**